File: classes/metamob.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

# Imports
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
import modules.globals as sg

# Class of a MetaMob (from MH FTP)
class METAMOB(sg.SqlAlchemyBase):

    # SQL Table Mapping
    __tablename__ = 'metamobs'
    # ID unique du metamonstre
    id = Column(Integer, primary_key=True)
    # Nom (sans modificateur)
    nom = Column(String(50))
    # Déterminant du monstre
    determinant = Column(String(10))
    # URL du blason
    blason_url = Column(String(150))
    
    # Associations Many-To-One
    mobs = relationship("MOB", back_populates="metamob")

    # Constructor is handled by SqlAlchemy, do not override

File: web/private/models/metamob.js
'use strict'; 

var sequelize = require('sequelize');
var MetamobTemplate = {};

/*
 * Definition
 */
MetamobTemplate.name = 'Metamob';
MetamobTemplate.table = 'metamobs';

MetamobTemplate.modelDefinition = {  
  id: {
    type: sequelize.INTEGER,
    primaryKey: true,
    allowNull: false
  },
  nom: {type: sequelize.STRING},
  determinant: {type: sequelize.STRING},
  blason_url: {type: sequelize.STRING}
};

MetamobTemplate.modelOptions = {
  name: {
    singular: 'metamob',
    plural: 'metamobs'
  }
};

module.exports = MetamobTemplate;
